src/v4/screen/CommunityHome/index.tsx

```diff
@@ -258,16 +258,24 @@ export default function CommunityHome({ route }: any) {
     );
   };
   const handleOnPressPostBtn = () => {
-    dispatch(
+    if (isJoin) {
+      return dispatch(
+        openPostTypeChoiceModal({
+          userId: (client as Amity.Client).userId as string,
+          targetId: communityId,
+          isPublic: communityData?.data.isPublic,
+          targetName: communityName,
+          targetType: PostTargetType.community,
+          postSetting: postSetting,
+          needApprovalOnPostCreation: (
+            communityData?.data as Record<string, any>
+          ).needApprovalOnPostCreation,
```

Comment:
Should we put `?` to mark it possible to be undefined? because as I see communityData is possible to undefined

```suggestion
          needApprovalOnPostCreation: (
            communityData?.data as Record<string, any>
          )?.needApprovalOnPostCreation,
```

Reply:
updated krub